### PR TITLE
[ALLUXIO-2952] Show both status code and error message for error responses from REST calls

### PR DESCRIPTION
**Pull request by uronce-cc**
With status code, clients for different languages based on the REST API can have more fine-grained exceptions.

### CODE REVIEW

**Comment by jsimsa** [comment]
please make alluxio-bot happy

File: core/server/common/src/main/java/alluxio/RestUtils.java [hunk 1]
```diff
@@ -85,20 +87,44 @@ private static Response createResponse(Object object) {
...
   }
 
   /**
-   * Creates an error response using the given message.
+   * Error response when {@link RestCallable#call()} throws an exception.
+   * It will be encoded into a Json string to be returned as an error response for the REST call.
+   */
+  private static class ErrorResponse {
+    private Status mStatus;
+    private String mMessage;
```
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RestUtils.java]: Should these be `final`?

File: core/server/common/src/main/java/alluxio/RestUtils.java [hunk 2]
```diff
@@ -85,20 +87,44 @@ private static Response createResponse(Object object) {
...
+      mMessage = message;
+    }
+
+    public Status getStatus() {
+      return mStatus;
+    }
+
+    public String getMessage() {
+      return mMessage;
+    }
```
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RestUtils.java]: Do we need the getters?

**Comment by aaudiber** [comment]
alluxio-bot, check this please

File: core/server/common/src/main/java/alluxio/RestUtils.java (hunk 1 above)
Comment (uronce-cc) [inline core/server/common/src/main/java/alluxio/RestUtils.java]: there are only getters

Comment (uronce-cc) [inline core/server/common/src/main/java/alluxio/RestUtils.java]: I'll add `final` in another PR

**Comment by uronce-cc** [comment]
@calvinjia will add a unit test for RestUtils in another PR.

File: core/server/common/src/main/java/alluxio/RestUtils.java (hunk 2 above)
Comment (uronce-cc) [inline core/server/common/src/main/java/alluxio/RestUtils.java]: it's needed for the `entity` method to serialize the java bean.

**Comment by uronce-cc** [comment]
Added a unit test in https://github.com/Alluxio/alluxio/pull/5766, may add an integration test to verify the encoded json strings in a further PR.

**Pull request merged**